test/unit/BCoWPool/BCoWPool_Verify.t.sol

```diff
@@ -0,0 +1,127 @@
+    correctOrder.sellToken = IERC20(makeAddr('unknown token'));
+    // it should revert
+    vm.expectRevert(IBPool.BPool_TokenNotBound.selector);
+    bCoWPool.verify(correctOrder);
+  }
+
+  function test_RevertWhen_OrderReceiverIsNotRECEIVER_SAME_AS_OWNERFlag() external {
+    correctOrder.receiver = makeAddr('somebodyElse');
+    // it should revert
+    vm.expectRevert(IBCoWPool.BCoWPool_ReceiverIsNotBCoWPool.selector);
+    bCoWPool.verify(correctOrder);
+  }
+
+  function test_RevertWhen_OrderIsValidForMoreThanMAX_ORDER_DURATION(uint256 _timeOffset) external {
+    _timeOffset = bound(_timeOffset, MAX_ORDER_DURATION + 1, type(uint32).max - block.timestamp);
+    correctOrder.validTo = uint32(block.timestamp + _timeOffset);
+    // it should revert
+    vm.expectRevert(IBCoWPool.BCoWPool_OrderValidityTooLong.selector);
+    bCoWPool.verify(correctOrder);
+  }
+
+  function test_RevertWhen_FeeAmountIsNotZero(uint256 _fee) external {
+    _fee = bound(_fee, 1, type(uint256).max);
+    correctOrder.feeAmount = _fee;
+    // it should revert
+    vm.expectRevert(IBCoWPool.BCoWPool_FeeMustBeZero.selector);
+    bCoWPool.verify(correctOrder);
+  }
+
+  function test_RevertWhen_OrderKindIsNotKIND_SELL(bytes32 _orderKind) external {
+    vm.assume(_orderKind != GPv2Order.KIND_SELL);
+    correctOrder.kind = _orderKind;
+    // it should revert
+    vm.expectRevert(IBCoWPool.BCoWPool_InvalidOperation.selector);
+    bCoWPool.verify(correctOrder);
+  }
+
+  function test_RevertWhen_TokenBalanceMarkerForBuyTokenIsNotDirectERC20Balances(bytes32 _balanceKind) external {
+    vm.assume(_balanceKind != GPv2Order.BALANCE_ERC20);
+    correctOrder.buyTokenBalance = _balanceKind;
+    // it should revert
+    vm.expectRevert(IBCoWPool.BCoWPool_InvalidBalanceMarker.selector);
+    bCoWPool.verify(correctOrder);
+  }
+
+  function test_RevertWhen_TokenBalanceMarkerForSellTokenIsNotDirectERC20Balances(bytes32 _balanceKind) external {
+    vm.assume(_balanceKind != GPv2Order.BALANCE_ERC20);
+    correctOrder.sellTokenBalance = _balanceKind;
+    // it should revert
+    vm.expectRevert(IBCoWPool.BCoWPool_InvalidBalanceMarker.selector);
+    bCoWPool.verify(correctOrder);
+  }
+
+  function test_RevertWhen_OrderBuyAmountExceedsMAX_IN_RATIO(uint256 _buyAmount) external {
+    _buyAmount = bound(_buyAmount, bmul(tokenInBalance, MAX_IN_RATIO) + 1, type(uint256).max);
+    correctOrder.buyAmount = _buyAmount;
+    // it should revert
+    vm.expectRevert(IBPool.BPool_TokenAmountInAboveMaxRatio.selector);
+    bCoWPool.verify(correctOrder);
+  }
+
+  function test_RevertWhen_ComputedTokenAmountOutIsLessThanOrderMinSellAmount() external {
+    correctOrder.sellAmount += 1;
+    // it should revert
+    vm.expectRevert(IBPool.BPool_TokenAmountOutBelowMinOut.selector);
+    bCoWPool.verify(correctOrder);
+  }
+
+  function test_WhenPreconditionsAreMet() external {
+    // it should return
+    bCoWPool.verify(correctOrder);
+  }
```

Comment:
since we checked for order `sellAmount + 1`, how would you feel here fuzzing from `0 -> calculatedOut` to not lose the branch where sellAmount is lesser than calculated (that is the same as when it's ==, but we're not showing that behaviour)

Reply:
seems reasonable, I don't think it's a different _branch_ since it covers the exact same code. As far as I understood, we can use a single testcase for different values if they don't cover different lines/branches of code. 

CCing @drgorillamd in case he considers we should use a separate testcase

Reply:
This makes sense to me, you have a doubt about it? I'm missing something?